sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/ViewDescriptorFactory.scala

```diff
@@ -206,6 +158,161 @@ private[impl] object ViewDescriptorFactory extends ComponentDescriptorFactory {
     ComponentDescriptor(nameGenerator, serviceName, component.getPackageName, kalixMethods, additionalMessages.toSeq)
   }
 
+  private def eventSourcedEntitySubscription(
+      component: Class[_],
+      tableType: Class[_],
+      tableName: String,
+      tableProtoMessageName: String) = {
+    var previousEntityClass: Option[Class[_]] = None
+
+    import ReflectionUtils.methodOrdering
+
+    def invalidComponentException(method: Method, extraMessage: String) = {
+      throw InvalidComponentException(
+        s"Method [${method.getName}] annotated with '@Subscribe.EventSourcedEntity' should either receive " +
+        "a single parameter of one of the event types or " +
+        s"two ordered parameters  of type [${tableType.getName}] and an event type. $extraMessage")
+    }
+
+    component.getMethods
+      .filter(hasEventSourcedEntitySubscription)
+      .sorted // make sure we get the methods in deterministic order
+      .map { method =>
+        // validate that all updates return the same type
+        val entityClass = method.getAnnotation(classOf[Subscribe.EventSourcedEntity]).value().asInstanceOf[Class[_]]
+        val stateClass = entityClass.getGenericSuperclass
+          .asInstanceOf[ParameterizedType]
+          .getActualTypeArguments
+          .head
+          .asInstanceOf[Class[_]]
+        previousEntityClass match {
+          case Some(entityClass) => // ok
+          case Some(other) =>
+            throw InvalidComponentException(
+              s"All update methods must return the same type, but [${method.getName}] returns [${entityClass.getName}] while a previous update method returns [${other.getName}]")
+          case None => previousEntityClass = Some(entityClass)
+        }
+
+        method.getParameterTypes.toList match {
+          case params if params.size > 2 =>
+            invalidComponentException(
+              method,
+              s"Subscription method should have one or two params, found ${params.size}")
+          case p1 :: Nil if p1 == tableType =>
+            invalidComponentException(method, "Subscription method only has the view type.")
+
+          case p1 :: p2 :: Nil if p1 != tableType =>
+            invalidComponentException(method, s"Subscription method first param must be view type [$tableType]")
+
+          case p1 :: p2 :: Nil if p1 == tableType && p2 == tableType =>
+            invalidComponentException(method, "Subscription method receives twice the view type [$tableType]")
+          case _ => // happy days, dev did good with the signature
+        }
```

Comment:
In retrospect, I think the idea of allowing the state to be part of the method was not so great. There are so many situations that users can get it wrong. I don't think that giving this option is helping the users. 

I think we should consider going to the original idea of having a `viewState` method.

Reply:
Yeah, kind of agree with that.. also, it would be consistent with the remaining component types.